cluster/gce/util.sh

```diff
@@ -2576,9 +2576,11 @@ function create-cloud-nat-router() {
 }
 
 function delete-all-firewall-rules() {
-  if fws=$(gcloud compute firewall-rules list --project "${NETWORK_PROJECT}" --filter="network=${NETWORK}" --format="value(name)"); then
+  if fws=$(gcloud compute firewall-rules list --project "${NETWORK_PROJECT}" --filter="network=${NETWORK}" --format='value[terminator=" "](name)'); then
     echo "Deleting firewall rules remaining in network ${NETWORK}: ${fws}"
-    delete-firewall-rules "$fws"
+    # Deliberately do not quote, do not change unless a bug is found
+    # shellcheck disable=SC2086
+    gcloud compute firewall-rules delete --project "${NETWORK_PROJECT}" --quiet ${fws}
```

Comment:
We don't need to make shellcheck unhappy, how about using pattern seen in this file, e.g. https://github.com/kubernetes/kubernetes/blob/1f07ce45325fc88da371b2b9b50e21623a5032a1/cluster/gce/util.sh#L3926-L3931

If you do something like
```
local -a fws
kube::util::read-array fws < <(gcloud compute firewall-rules list --project "${NETWORK_PROJECT}" --filter="network=${NETWORK}" --format="value(name)")
if (( "${#fws[@]}" > 0 )); then 
 echo "Deleting firewall rules remaining in network ${NETWORK}: ${fws}"
 delete-firewall-rules "${fws[@]}"
fi

this should probably work


```

Reply:
Hi Maciej, I'm not sure, if I understand your comment.

The main point behind this PR is not to use the `delete-firewall-rules` func, which might not be as stable as it looks. Instead, I want to delete all firewalls at once with native `gcloud` command.
The linter issue is about `"${fws[@]}"` vs `${fws[@]}` - it has to be the 2nd option, as we're passing a inline, space separated list of firewalls names, e.g. `fw1 fw2 fw3 ...` - if you add quotation marks, it thinks, it's a single firewall name with spaces in a middle, which is not valid.

This pattern is used in many places, like flags:
https://github.com/kubernetes/kubernetes/blob/8a0e0c7f79d6d5c9b4acb0accbcb5bbee5cc9ef4/cluster/gce/gci/configure-helper.sh#L2380-L2381

Now, your suggestion is more around `kube::util::read-array` vs `format='value[terminator=" "](name)'`, which I don't have a strong opinion about and I'm happy to refactor, if you think it's cleaner.

Reply:
Actually, looking closer at it, `kube::util::read-array` it's still an array, which you can iterate on (or e.g. check, it's size). I don't think it would work in this case - unless I don't see something?

Reply:
> The main point behind this PR is not to use the delete-firewall-rules func, which might not be as stable as it looks. Instead, I want to delete all firewalls at once with native gcloud command.

OK, I think this is where we have misunderstanding.

IIUC the real bug here is that we call delete-firewall-rules with all firewalls in a single argument, which makes this call no-op, as there is no firewall with such name (there are no "Deleted [https://www.googleapis.com/compute/v1/projects/k8s-infra-e2e-boskos-scale-17/global/firewalls/XXX]" lines in the log which are printed by gcloud when called to delete object).

IMHO we should fix this bug. There is no need to migrate to custom gcloud call.

Reply:
Ok, I see your point now.

All it all, I think it's all about calling `delete-firewall-rules`, which requires list of arguments (vs a single argument which is a list) - I think that was a main issue.

I'd go with your approach, as it requires less changes to the actual logic.